### PR TITLE
fix: Crash on new instance from previous installation

### LINKED ISSUE #1266: [Bug] Romm does not correctly reject invalid session keys
**RomM version**
rommapp/romm:latest

**Describe the bug**
Romm does not check if a session key is invalid and reset, instead shows a corrupted interface

**To Reproduce**

1. set up romm and create a user account, then log in
2. reset romm, or set up romm on a new server that points to the same url
2a. Theoretically make a fake session key and pose as a hacker attempting to illegally access romm
4. Romm will launch to the full interface, but with all user specific sections blank. logging out will not redirect to the setup page, but opening the site in a different browser with no previous cookies or cache will launch the setup page

**Expected behavior**
romm should redirect to the login page (or a setup page if not previously set up) whenever an invalid session key is given by the user

### PR DESCRIPTION
**Pull request by zurdi15**
When setting up a new installation, if the browser has old cookies (romm_session cookie specifically) from an old instance or from an user that no longer exists in the database, RomM crashes without recover. This fix adds a check in the middleware to avoid the crash and clean the session cookies gracefully.

Closes #1266

### CODE REVIEW

File: backend/handler/auth/middleware.py [hunk 1]
```diff
@@ -107,7 +109,15 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
...
                 scope["session"] = jwt_claims
                 initial_session_was_empty = False
-            except BadSignatureError:
+
+                # Check if the user exists
+                db_user_handler = DBUsersHandler()
+                user = db_user_handler.get_user_by_username(jwt_claims.get("sub"))
+                if not user:
+                    scope["session"] = {}
+                    initial_session_was_empty = True
```
Comment (gantoine) [inline backend/handler/auth/middleware.py]: making a db request on every request is very expensive, and i think there's a better way to handle this (though i cant think of it right now)...

Comment (zurdi15) [reply backend/handler/auth/middleware.py]: Yeah I know, we need to figure out this one

Comment (zurdi15) [reply backend/handler/auth/middleware.py]: maybe we can make the db request only in the heartbeat instead on every request? that will reduce the database requests but will clear session when refreshing the page if the user no longer exists

Comment (zurdi15) [reply backend/handler/auth/middleware.py]: i tested it and it won't work, if any requests is made before the heartbeat it will crash

Comment (zurdi15) [reply backend/handler/auth/middleware.py]: Okay fixed! those changes should do the trick. I tested removing and/or disabling users with an existing session cookie and works like a charm without requesting the database

File: backend/handler/auth/middleware.py [hunk 2]
```diff
@@ -107,7 +108,8 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
                 jwt_claims = self._validate_jwt_payload(jwt_payload)
                 scope["session"] = jwt_claims
                 initial_session_was_empty = False
-            except BadSignatureError:
+
+            except (BadSignatureError, HTTPException):
```
Comment (gantoine) [inline backend/handler/auth/middleware.py]: i don't think any calls in here will throw an `HTTPException`, do you can probably remove it.

**Review by gantoine — APPROVED** [review]
yeah good fix!

File: backend/handler/auth/middleware.py (hunk 2 above)
Comment (zurdi15) [reply backend/handler/auth/middleware.py]: You are right!

**Pull request merged**